test/1_stdlib/Collection.swift

```diff
@@ -202,5 +202,41 @@ func testIsEmptyFirstLast() {
 }
 testIsEmptyFirstLast()
 
+/// A `CollectionType` that vends just the default implementations for
+/// `CollectionType` methods.
+struct CollectionOnly<T: CollectionType> : CollectionType {
+  var base: T
+
+  var startIndex: T.Index {
+    return base.startIndex
+  }
+
+  var endIndex: T.Index {
+    return base.endIndex
+  }
+
+  func generate() -> T.Generator {
+    return base.generate()
+  }
+
+  subscript(position: T.Index) -> T.Generator.Element {
+    return base[position]
+  }
+}
+
+func testFirstPerformance() {
+  // accessing `first` should not perform duplicate work on lazy collections
+  let col_ = (0..<10).lazy.filter({ print("filter: \($0)"); return (2..<8).contains($0) })
```

Comment:
@kballard you didn’t know this, but we are very intentionally avoiding adding any new FileCheck testing code to the standard library suite, and when we get time, we rewrite FileCheck tests to use StdlibUnittest.